qiskit/transpiler/passes/layout/dense_layout.py

```diff
@@ -35,39 +35,33 @@ class DenseLayout(AnalysisPass):
         by being set in `property_set`.
     """
 
-    def __init__(self, coupling_map, backend_prop=None):
+    def __init__(self, coupling_map, backend_prop=None, target=None):
```

Comment:
Is there a good way we can make `coupling_map` optional when `target` is provided? I'm guessing the answer is "no", but just curious.

Reply:
I can set a default on `coupling_map` to `None`. I did this other passes, not sure why I missed this here. The pass already supports a user explicitly setting `coupling_map=None` in the constructor (but it'll fail at runtime). You'll be required to set one or the other or it will fail at runtime, which isn't any different that manually setting `coupling_map` to `None` today

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7776/commits/5814d0249bdc88ac7b5675694f1ee273a0730d6c